tasks/eth/008-fp-upgrade/NestedSignFromJson.s.sol

```diff
@@ -0,0 +1,263 @@
+            require(
+                accountAccess.oldBalance == accountAccess.newBalance,
+                string.concat("Unexpected balance change: ", vm.toString(accountAccess.account))
+            );
+            require(
+                accountAccess.kind != VmSafe.AccountAccessKind.SelfDestruct,
+                string.concat("Self-destructed account: ", vm.toString(accountAccess.account))
+            );
+
+            for (uint256 j; j < accountAccess.storageAccesses.length; j++) {
+                Vm.StorageAccess memory storageAccess = accountAccess.storageAccesses[j];
+
+                address account = storageAccess.account;
+                require(account != proxies.L1CrossDomainMessenger, "state-000");
+                require(account != proxies.L1ERC721Bridge, "state-100");
+                require(account != proxies.L1StandardBridge, "state-200");
+                require(account != proxies.OptimismMintableERC20Factory, "state-300");
+            }
+        }
+    }
+
+    function checkSemvers() internal view {
+        // These are the expected semvers based on the `op-contracts/v1.4.0-rc.4` release.
+        // https://github.com/ethereum-optimism/optimism/releases/tag/op-contracts%2Fv1.4.0-rc.4
+        require(ISemver(proxies.L1CrossDomainMessenger).version().eq("2.3.0"), "semver-100");
+        require(ISemver(proxies.L1ERC721Bridge).version().eq("2.1.0"), "semver-700");
+        require(ISemver(proxies.L1StandardBridge).version().eq("2.1.0"), "semver-200");
+        require(ISemver(proxies.OptimismMintableERC20Factory).version().eq("1.9.0"), "semver-400");
+        require(ISemver(proxies.OptimismPortal).version().eq("3.10.0"), "semver-500");
+        require(ISemver(proxies.SystemConfig).version().eq("2.2.0"), "semver-600");
+        require(ISemver(proxies.ProtocolVersions).version().eq("1.0.0"), "semver-800");
+        require(ISemver(proxies.SuperchainConfig).version().eq("1.1.0"), "semver-900");
+    }
+
+    /// @notice Asserts that the SystemConfig is setup correctly
+    ///         Assertions are similar to the 006-MCP-L1 upgrade
+    function checkSystemConfig() internal view {
+        console.log("Running chain assertions on the SystemConfig");
+
+        require(proxies.SystemConfig.code.length != 0, "200");
+        require(EIP1967Helper.getImplementation(proxies.SystemConfig).code.length != 0, "201");
+
+        SystemConfig configToCheck = SystemConfig(proxies.SystemConfig);
+        require(configToCheck.owner() == superchainConfigGuardian, "300");
+        require(configToCheck.overhead() == gasPriceOracleOverhead, "400");
+        require(configToCheck.scalar() == gasPriceOracleScalar, "500");
+        require(configToCheck.batcherHash() == bytes32(uint256(uint160(batchSenderAddress))), "600");
+        require(configToCheck.gasLimit() == uint64(l2GenesisBlockGasLimit), "700");
+        require(configToCheck.unsafeBlockSigner() == p2pSequencerAddress, "800");
+
+        // TODO: solc is acting up
```

Comment:
lol, what's it doing?

Reply:
fixed. (solc works in mysterious ways. Importing `ResourceConfig` directly somehow fixes a compiler issue)